### PR TITLE
Add autofix to unit-case

### PR DESCRIPTION
**Pull request by hiroppy**
> Which issue, if any, is this issue related to?

https://github.com/stylelint/stylelint/issues/2829

e.g. "#000" or "None, as it's a documentation fix."

> Is there anything in the PR that needs further explanation?

## lower
```
a { width: 10PIXEL; } -> a { width: 10pixel; }
a { width: calc(10PX * 2); } -> a { width: calc(10px * 2); }
```

## uppter
```
a { width: 10pixel; } -> a { width: 10PIXEL; }
a { width: calc(10px * 2); } -> a { width: calc(10PX * 2); }
```

e.g. "No, it's self explanatory."

### CODE REVIEW

File: lib/rules/unit-case/__tests__/index.js [hunk 1]
```diff
@@ -666,3 +666,51 @@ testRule(rule, {
...
+    },
+    {
+      code: "a { width: calc(10PX * 2); }",
+      fixed: "a { width: calc(10px * 2); }",
+      message: messages.expected("PX", "px")
+    }
+  ]
+});
+
+testRule(rule, {
```
Comment (evilebottnawi) [inline lib/rules/unit-case/__tests__/index.js]: need tests around scss/less/css variables

Comment (abouthiroppy) [reply lib/rules/unit-case/__tests__/index.js]: added tests.

**Comment by jeddy3** [comment]
@hiroppy Friendly ping to see if you're still working on this PR?

File: lib/rules/unit-case/index.js [hunk 2]
```diff
@@ -57,6 +58,23 @@ const rule = function(expectation) {
...
+
+            const res = `${stringStart}${expectedUnit}${stringEnd}`;
+
+            if (node.name === "media") {
+              node.params = res;
+            } else {
+              node.value = res;
+            }
+          });
+        }
```
Comment (hudochenkov) [inline lib/rules/unit-case/index.js]: Move it before `report` and add `return` at the end of `if` statement. This way we won't show rule violation when we fixed the problem.

**Comment by hiroppy** [comment]
@hudochenkov updated!

**Comment by jeddy3** [comment]
Can we not use `postcss-value-parser` to update the AST, rather than use `stylesearch`?

**Comment by hudochenkov** [comment]
@jeddy3 you mean, could we _use_ `postcss-value-parser` instead of `stylesearch`?

**Comment by jeddy3** [comment]
Oops, yep... that wasn't very clear.

Yes, I mean can we use `postcss-value-parser` to update the AST?

**Comment by jeddy3** [comment]
@hiroppy Do you think you'll be able to update this PR to use `postcss-value-parser` to update the AST, rather than use `stylesearch`. Updating the more granular values AST will likely be a more robust solution.

**Comment by hiroppy** [comment]
@jeddy3 Thank you for the suggestion. OK, I will try using postcss-value-parser :). However, I am a little busy this week so please wait a moment.

**Comment by yepninja** [comment]
Already solved #3725

**Pull request closed without merging**